template/src/search/providers.ts

```diff
@@ -292,7 +292,38 @@ export function createProviders(
         if (squirrelDocumentHighlights) {
             return squirrelDocumentHighlights
         }
-        return null
+        const contentAndToken = await getContentAndToken(textDocument, position)
+        if (!contentAndToken) {
+            return null
+        }
+        const { text, searchToken } = contentAndToken
+        const result: sourcegraph.DocumentHighlight[] = []
+        let line = 0
+        let lastLineIndex = 0
+        const tokenPattern = new RegExp('\\b' + searchToken + '\\b', 'g')
+        const matches = [...text.matchAll(tokenPattern)]
+        // TODO: why is matches empty?
```

Comment:
I don't understand why `matches` is always empty here. Any advice on what I'm doing wrong? When I test the same regexp manually then it works as expected, it's only empty when running via webworkers 🤔

Reply:
Do you know the values of text and pattern here at runtime?

Reply:
Yes. I even used "copy object" on the actual logged values and `matchAll()` worked as expected against that object in the JS console. The `text` value has the full multi-line string value of the document and the pattern is correct too.